lib/knex-builder/make-knex.js

```diff
@@ -141,10 +141,8 @@ function initContext(knexFn) {
         const trx = this.client.transaction(container, config, outerTx);
         return trx;
       } else {
-        return new Promise((resolve, reject) => {
-          const trx = this.client.transaction(resolve, config, outerTx);
-          trx.catch(reject);
```

Comment:
I don't see this catch blocked added elsewhere, was its removal intentional?

Reply:
I don't really get why this Promise is here, when in the other branch it just returns the trx element, so ideally I'd find a way to remove it, based on the tests failing, that might not be possible :/

Reply:
Saying that our transaction logic is messy and complicated would be an... understatement. There are lots of edge cases there.